test/test.js

```diff
@@ -330,6 +330,30 @@ describe('Page', function() {
     }));
   });
 
+  describe('ExecutionContext.queryObjects', function() {
+    it('should work', SX(async function() {
+      // Instantiate an object
+      await page.evaluate(() => window.map = new Map());
+      const prototypeHandle = await page.evaluateHandle(() => Map.prototype);
+      const objectsHandle = await page.queryObjects(prototypeHandle);
+      const count = await page.evaluate(objects => objects.length, objectsHandle);
```

Comment:
To make me feel safer, can we actually check that the object here is the map and is usable.

Reply:
Done.